### PR TITLE
[scs] Vectors representation

### PR DESCRIPTION
**Pull request by NikitaZotov**
* [x] Read PR [documentation](https://github.com/ostis-ai/sc-machine/blob/main/docs/dev/pr.md/)
* [x] Update changelog
* [x] Update documentation

Now you can represent oriented sets in SCs in this form: `<a, b, c>`.

### CODE REVIEW

File: sc-memory/tests/scs/units/test_scs_level_4.cpp [hunk 1]
```diff
@@ -91,3 +91,52 @@ TEST(scs_level_4, simple_2)
...
+      },
+  });
+
+  auto const & triples = parser.GetParsedTriples();
+
+  EXPECT_EQ(triples.size(), 6u);
+
+  EXPECT_EQ(triples[1].m_target, triples[0].m_edge);
+  EXPECT_EQ(triples[3].m_target, triples[2].m_edge);
+}
```
Comment (AlexandrZagorskiy) [inline sc-memory/tests/scs/units/test_scs_level_4.cpp:142]: add more tests. Can vector be empty? What about vector from a single element? Can vector contain another vector?

File: sc-memory/tests/scs/units/test_scs_level_4.cpp [hunk 2]
```diff
@@ -91,3 +91,52 @@ TEST(scs_level_4, simple_2)
   EXPECT_EQ(triples[2].m_edge, triples[3].m_target);
   EXPECT_EQ(triples[2].m_edge, triples[4].m_target);
 }
+
+TEST(scs_level_4, vector_simple_1)
```
Comment (AlexandrZagorskiy) [inline sc-memory/tests/scs/units/test_scs_level_4.cpp:95]: Why it is a scs_level_4? Why not a 6 level, for example

**Review by AlexandrZagorskiy — CHANGES_REQUESTED** [review]

File: sc-memory/tests/scs/units/test_scs_level_4.cpp (hunk 1 above)
Comment (NikitaZotov) [reply sc-memory/tests/scs/units/test_scs_level_4.cpp:269]: Fixed

**Review by AlexandrZagorskiy — APPROVED** [review]

File: sc-memory/sc-memory/scs/scs.g4 [hunk 3]
```diff
@@ -222,41 +222,67 @@ idtf_edge returns [ElementHandle handle]
...
+      m_parser->ProcessTriple(typeClass, typeEdge, $ctx->handle);
+    }
+  a1=attr_list? i1=idtf_common
+    {
+      ElementHandle const edge = m_parser->ProcessConnector("->");
+      m_parser->ProcessTriple($ctx->handle, edge, $ctx->i1->handle);
+
+      if ($ctx->a1 != nullptr)
+      {
+        APPEND_ATTRS($ctx->a1->items, edge);
```
Comment (AlexandrZagorskiy) [inline sc-memory/sc-memory/scs/scs.g4:247]: can we make naming more clear? Or such refactor should be in another PR?

**Review by AlexandrZagorskiy — APPROVED** [review]

File: sc-memory/sc-memory/scs/scs.g4 (hunk 3 above)
Comment (NikitaZotov) [reply sc-memory/sc-memory/scs/scs.g4:247]: https://github.com/ostis-ai/sc-machine/issues/196
